agent/stats/task_windows.go

```diff
@@ -0,0 +1,179 @@
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package stats
+
+import (
+	"context"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/stats/resolver"
+
+	dockerstats "github.com/docker/docker/api/types"
+	"github.com/pkg/errors"
+)
+
+// Making it visible for unit testing
+var execCommand = exec.Command
+
+type StatsTask struct {
+	*statsTaskCommon
+}
+
+func newStatsTaskContainer(taskARN string, containerPID string, numberOfContainers int,
+	resolver resolver.ContainerMetadataResolver, publishInterval time.Duration, taskENIs task.TaskENIs) (*StatsTask, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	devices := make([]string, len(taskENIs))
+	for count, device := range taskENIs {
+		devices[count] = device.LinkName
+	}
+
+	return &StatsTask{
+		statsTaskCommon: &statsTaskCommon{
+			TaskMetadata: &TaskMetadata{
+				TaskArn:          taskARN,
+				ContainerPID:     containerPID,
+				DeviceName:       devices,
+				NumberContainers: numberOfContainers,
+			},
+			Ctx:                   ctx,
+			Cancel:                cancel,
+			Resolver:              resolver,
+			metricPublishInterval: publishInterval,
+		},
+	}, nil
+}
+
+func (taskStat *StatsTask) retrieveNetworkStatistics() (map[string]dockerstats.NetworkStats, error) {
+	if len(taskStat.TaskMetadata.DeviceName) == 0 {
+		return nil, errors.Errorf("unable to find any device name associated with the task %s", taskStat.TaskMetadata.TaskArn)
+	}
+
+	networkStats := make(map[string]dockerstats.NetworkStats, len(taskStat.TaskMetadata.DeviceName))
+	for _, device := range taskStat.TaskMetadata.DeviceName {
+		networkAdaptorStatistics, err := taskStat.getNetworkAdaptorStatistics(device)
+		if err != nil {
+			return nil, err
+		}
+		networkStats[device] = *networkAdaptorStatistics
+	}
+
+	return networkStats, nil
+}
+
+// getNetworkAdaptorStatistics returns the network statistics per container for the given network interface.
+func (taskStat *StatsTask) getNetworkAdaptorStatistics(device string) (*dockerstats.NetworkStats, error) {
+	// Ref: https://docs.microsoft.com/en-us/powershell/module/netadapter/get-netadapterstatistics?view=windowsserver2019-ps
+	// The Get-NetAdapterStatistics cmdlet gets networking statistics from a network adapter.
+	// The statistics include broadcast, multicast, discards, and errors.
+	cmd := "Get-NetAdapterStatistics -Name \"" + device + "\" | Format-List -Property *"
+	out, err := execCommand("powershell", "-Command", cmd).CombinedOutput()
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to run Get-NetAdapterStatistics for %s", device)
+	}
+	str := string(out)
+
+	lines := strings.Split(str, "\n")
```

Comment:
I think the test string below will have only \n because of the go formatter but will PowerShell output `\r\n`?

Reply:
Yes Powershell would output `\r\n` but we are using `strings.TrimSpace` before adding the key and value into the map. This method would remove any additional spaces and `\r` as well.

Reply:
I did not know that `\r` was considered whitespace! Just verified this is true. Cool!